File: daily-practice/sum_numbers.py
def sum_numbers(nums):
    # 1. 합계를 저장할 변수 total을 0으로 초기화하세요.
    total = 0

    # 2. nums의 숫자를 하나씩 꺼내 total에 더하세요.
    for i in nums:
        total += i
    # 3. 최종 합계를 반환하세요.
    return total


if __name__ == "__main__":
    print(sum_numbers([1, 2, 3, 4]))  # 예상 결과: 10
    print(sum_numbers([]))            # 예상 결과: 0
    print(sum_numbers([-2, 5, 1]))    # 예상 결과: 4

File: 06-base-ps-java/src/main/java/com/algopatterns/easy/SumNumbers.java
package com.algopatterns.easy;

public class SumNumbers {

    public static int sumNumbers(int[] nums) {
        // 1. 합계를 저장할 변수 total을 0으로 초기화하세요.
        int total = 0;
        // 2. nums의 숫자를 하나씩 꺼내 total에 더하세요.
        for (int i : nums) {
            total += i;
        }
        // 3. 최종 합계를 반환하세요.
        return total;
    }

    public static void main(String[] args) {
        System.out.println(sumNumbers(new int[]{1, 2, 3, 4})); // 예상 결과: 10
        System.out.println(sumNumbers(new int[]{}));            // 예상 결과: 0
        System.out.println(sumNumbers(new int[]{-2, 5, 1}));    // 예상 결과: 4
    }
}
